context/logger.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package context
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+var (
+	// G is an alias for GetLogger.
+	//
+	// We may want to define this locally to a package to get package tagged log
+	// messages.
+	G = GetLogger
+
+	// LogEntry provides a public and standard logger instance.
+	LogEntry = logrus.NewEntry(logrus.StandardLogger())
+)
+
+type (
+	loggerKey struct{}
+
+	stderrHook struct{}
```

Comment:
@stevvooe 
In this design, I set default logger.Out to stdout, because I want to print info and debug to stdout. 
```go
LogEntry.Logger.Out = os.Stdout
```
I create hook because I can fire the output to stderr, by `hook.Fire()` function. And I trigger it when level = [fatal,panic,error,warn], as `hook.Level()` is doing. Hook is necessary for logger.

About if we should use logger, I mostly agree with @vbatts. If we use logrus directly without logger, maybe the applicability will be a little narrow (although I can not figure out all usage now).